frontend/src/views/admin/matching/applicant-view.tsx

```diff
@@ -0,0 +1,666 @@
+import React from "react";
+import classNames from "classnames";
+
+import { Position, Application } from "../../../api/defs/types";
+import { ApplicantSummary, Match } from "./types";
+
+import { FaFilter, FaTable, FaTh, FaLock } from "react-icons/fa";
+import { BsInfoCircleFill, BsStarFill } from "react-icons/bs";
+import { RiStickyNoteFill } from "react-icons/ri";
+
+import ToggleButton from "react-bootstrap/ToggleButton";
+import ToggleButtonGroup from "react-bootstrap/ToggleButtonGroup";
+import Collapse from 'react-bootstrap/Collapse';
+import { Form, Table, Dropdown, Modal, Button } from "react-bootstrap";
+
+import { sum } from "../../../api/mockAPI/utils";
+import { round } from "../../../libs/utils";
+
+import { upsertMatch } from "./actions";
+import { useThunkDispatch } from "../../../libs/thunk-dispatch";
+
+import {
+    getApplicantMatchForPosition,
+    getPositionPrefForPosition,
+    getApplicantTotalHoursAssigned,
+} from "./utils";
+import { SortDropdowns, applySorts } from "./sorts";
+import { FilterModal, applyFilters, FilterListItem } from "./filters";
+
+import { ApplicationDetails } from "../applications/application-details";
+
+import "./styles.css";
+
+const statusMapping: Record<string, string[]> = {
+    "Assigned": ["assigned"],
+    "Assigned (Staged)": ["staged-assigned"],
+    "Starred": ["starred"],
+    "Applied": ["applied"],
+    "Hidden": ["hidden"],
+};
+
+export function ApplicantView({
+    position,
+    applicants,
+    setMarkAsUpdated,
+}: {
+    position: Position | null;
+    applicants: ApplicantSummary[];
+    setMarkAsUpdated: Function;
```

Comment:
`setMarkAsUpdated` -> `markAsUpdated` or `setUpdated` or `setUpdatedFlag`. Something that makes sense in English. Unless I don't understand what this function is supposed to do, in which case, other code should probably be reworked to remove it.